src/sagemaker/estimator.py

```diff
@@ -920,21 +930,30 @@ def deploy(
                 Implementations may customize ``create_model()`` to accept
                 ``**kwargs`` to customize model creation during deploy.
                 For more, see the implementation docs.
-
+        Raises:
+             ValueError: If serverless inference config is not specified and instance type
+                and instance count are also not specified
```

Comment:
why do we need this?

Reply:
Cause the way we’re doing this, is to let customers pass an empty config object to let us know it’s a serverless endpoint to be deployed. We only default values if they pass an empty object. If they just leave that arg as None, we’ll not using serverless endpoint but normal endpoint. So that means, we need to check to ensure instance_type/instance_count and serverless_inference_config cannot be None in the same time.